### ISSUE TITLE
Support import of Open XML Strict XSLX files

### ISSUE THREAD

**Issue by msaby** [OpenRefine/OpenRefine#2221]
**Describe the bug**

Impossible to import a XLSX file made with the last Excel version
My Excel 365 Desktop version : 
![image](https://user-images.githubusercontent.com/932804/69142831-eda4ed80-0ac7-11ea-8829-295f9f325f0b.png)

Error message : Attempted to parse as an Excel file but failed. Invalid XML.

![image](https://user-images.githubusercontent.com/932804/69142580-6788a700-0ac7-11ea-8c61-c4ebe755a85a.png)

Logs : 12:21:59.748 [..importers.ExcelImporter] Error generating parser UI initialization data for Excel file - invalid XML (181ms)
org.apache.poi.ooxml.POIXMLException: Strict OOXML isn't currently supported, please see bug #57699
        at org.apache.poi.ooxml.POIXMLDocumentPart.getPartFromOPCPackage(POIXMLDocumentPart.java:750)
        at org.apache.poi.ooxml.POIXMLDocumentPart.<init>(POIXMLDocumentPart.java:148)
        at org.apache.poi.ooxml.POIXMLDocumentPart.<init>(POIXMLDocumentPart.java:138)
        at org.apache.poi.ooxml.POIXMLDocument.<init>(POIXMLDocument.java:61)
        at org.apache.poi.xssf.usermodel.XSSFWorkbook.<init>(XSSFWorkbook.java:276)
        at org.apache.poi.xssf.usermodel.XSSFWorkbook.<init>(XSSFWorkbook.java:303)
        at com.google.refine.importers.ExcelImporter.createParserUIInitializationData(ExcelImporter.java:100)
        at com.google.refine.importing.DefaultImportingController.doInitializeParserUI(DefaultImportingController.java:216)
        at com.google.refine.importing.DefaultImportingController.doPost(DefaultImportingController.java:91)
        at com.google.refine.commands.importing.ImportingControllerCommand.doPost(ImportingControllerCommand.java:68)
        at com.google.refine.RefineServlet.service(RefineServlet.java:189)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:820)
        at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
        at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
        at org.mortbay.servlet.UserAgentFilter.doFilter(UserAgentFilter.java:81)
        at org.mortbay.servlet.GzipFilter.doFilter(GzipFilter.java:132)
        at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
        at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
        at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
        at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
        at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
        at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
        at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
        at org.mortbay.jetty.Server.handle(Server.java:326)
        at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
        at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
        at org.mortbay.jetty.HttpParser.parseNext(HttpParser.java:547)
        at org.mortbay.jetty.HttpParser.parseAvailable(HttpParser.java:212)
        at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
        at org.mortbay.jetty.bio.SocketConnector$Connection.run(SocketConnector.java:228)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)

**To Reproduce**

**Current Results**
What results occured or were shown.

**Expected behavior**
A clear and concise description of what you expected to happen or to show.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Win10
 - Browser Version:Chrome
 - JRE or JDK Version  JRE 1.8.0_231

**OpenRefine (please complete the following information):**
 - Version : Openrefine 3.3 beta

**Datasets**
If you are allowed and are OK with making your data public, it would be awesome if you can include or attach the data causing the issue or a URL pointing to where the data is.
If you are concerned about keeping your data private, ping us on our [mailing list](https://groups.google.com/forum/#!forum/openrefine)

**Additional context**

**Comment by msaby** [comment]
It seems that I used inadvertently this option in Excel, for using "strict mode". This makes the file unreadble because of that old bug in POI product https://bz.apache.org/bugzilla/show_bug.cgi?id=57699 . 
![image](https://user-images.githubusercontent.com/932804/69143130-98b5a700-0ac8-11ea-83c3-49f6b7850e68.png)

Hopefully I can save it as an ordinary Excel file, but if strict mode Excel is not supported yet by Openrefine it should be documented

**Comment by thadguidry** [comment]
"Strict" means that it adheres to the ISO/IEC standardization (does not contain Microsoft-specific data types, which is a good thing when you are looking for compatibility between tools for Read/Write/Edit).  So "Strict" is a better choice when you have it.  Generally, [ECMA-376](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-376,%20Second%20Edition,%20Part%203%20-%20Markup%20Compatibility%20and%20Extensibility.zip)
For historical context, I'd suggest reading these...
https://www.microsoft.com/en-us/microsoft-365/blog/2012/08/13/new-file-format-options-in-the-new-office/
https://docs.microsoft.com/en-us/office/open-xml/introduction-to-markup-compatibility

Now, since OpenRefine relies entirely on Apache POI to do the heavy lifting for MS Office file handling, we have to ensure we are always using the latest release of POI and making sure our tests continue to pass and validate changes in any standards (basially, upstream testing POI supported formats that WE support or want to support)

It also means that Apache POI team needs to keep up with OOXML changes 

From POI: 

> You'd use HSSF if you needed to read or write an Excel file using Java (XLS).
> You'd use XSSF if you need to read or write an OOXML Excel file using Java (XLSX).
> The combined SS interface allows you to easily read and write all kinds of Excel files (XLS and XLSX) using Java.
> Additionally there is a specialized SXSSF implementation which allows to write very large Excel (XLSX) files in a memory optimized way.

We actually use all of those interfaces, except SXSSF.
https://github.com/OpenRefine/OpenRefine/blob/master/main/src/com/google/refine/importers/ExcelImporter.java

Which ultimately means that POI XSSF interface needs to support "Strict" directly and not down convert to non-strict.
That will be work for the POI team to do (or any volunteer) in https://bz.apache.org/bugzilla/show_bug.cgi?id=57699

In doing the work within POI, essentially someone would need to produce mapping for where Office Open XML File Formats differs from the ISO/IEC-29500:2012 standards which is detailed here:
https://docs.microsoft.com/en-us/openspecs/office_standards/ms-oi29500/1fd4a662-8623-49c0-82f0-18fa91b413b8

**Comment by thadguidry** [comment]
@pjfanning Hi, So with the new PR merged at https://github.com/pjfanning/excel-streaming-reader/pull/21 to support Strict OOXML do you think we are unblocked here? What happens at the POI level now with this work done?
@tfmorris I guess @pjfanning implementation is sorta compatible for us if we used only for streaming import of large XLSX files? https://github.com/pjfanning/excel-streaming-reader#implementation-details

I'll let you 2 discuss it here.

**Comment by pjfanning** [comment]
I am on the PMC for Apache POI and there is no plan to support Strict mode. Just too much work and noone has the time or the urgent need to implement it. Code Contributions would be considered.

https://github.com/pjfanning/excel-streaming-reader/pull/21 is just experimental code and I have little interest in doing further work on it. There is very little test coverage. I will look at PRs if they come in but I'm not expecting any.

**Comment by tfmorris** [comment]
I've created #3407 to improve the error handling and message and we already have #3085 to improve the documentation.

I'm closing this as Won't Fix since there's no prospect that it'll happen in the foreseeable future.

**Comment by pjfanning** [comment]
@tfmorris this format is now partially supported in https://github.com/pjfanning/excel-streaming-reader - there is at least one issue to fix but that is in progress

**Comment by tfmorris** [comment]
Since progress is being made upstream, reopening to track that work for any of our users interested in OOXML Strict format support.
